### ISSUE TITLE
Sources are not in Sync with class files (in Maven central and IntelliJ)

### ISSUE THREAD

**Issue by uhafner** [TNG/ArchUnit#636]
In IntelliJ I get the following error message when trying to download the Java source code of your library: 

```
Library source does not match the bytecode for class
```

I'm using a maven project that resolves the artifact from Maven central. I used the *Download Sources* option to download the sources. Is this due to the Maven shade plugin? Or is this an IntelliJ Bug (like https://youtrack.jetbrains.com/issue/IDEA-265862)?

I included your library as:
```xml 
     ...
    <archunit.version>0.20.0</archunit.version>
    ...
    <dependency>
      <groupId>com.tngtech.archunit</groupId>
      <artifactId>archunit-junit5</artifactId>
      <version>${archunit.version}</version>
      <scope>test</scope>
    </dependency>
```

IntelliJ then complains and shows the class diff:

![Bildschirmfoto 2021-07-15 um 14 19 59](https://user-images.githubusercontent.com/503338/125787531-c2d63ecf-7f5a-4e22-9302-2316f45eb83f.png)

**Comment by uhafner** [comment]
It seems that only some classes have problems. E.g., I can open the source code for `ClassFileImporter` but not for `ArchConfiguration`.

**Comment by hankem** [comment]
I see the same behavior for `JavaClass` (using IntelliJ IDEA 2021.1.3).  
`ArchConfiguration` however "works" for me in the same way as `JavaClass`, i.e. shows a similar diff.

Apart from code style differences (e.g. static imports or field initializers) that a decompiler just cannnot reconstruct from byte code, I believe the only real differences are [relocated packages](https://github.com/TNG/ArchUnit/blob/v0.20.0/build-steps/archiving/archiving.gradle#L41-L43), such as `com.google.common` vs. `com.tngtech.archunit.thirdparty.com.google.common`. 

I must admit that I don't know what is expected in a `sources.jar`. From the point of view of an IDE, I totally understand that relocated source code would be desirable, and from a quick look it seems that e.g. the [Maven Shade Plugin has an option for that](https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#shadeSourcesContent).

**Comment by codecholeric** [comment]
Yes, I think the only reason IntelliJ complains is because Guava and ASM are relocated, so the source code has different imports than the bytecode. However we're using the Gradle Shadow plugin, not the Maven Shade plugin, and I couldn't find any option when I looked at it. 
Also I don't know what is correct here, because actually it's true, the source code doesn't match the bytecode, but it's also expected behavior and no real problem (apart from the ugly looks inside of IntelliJ :thinking:)
